modules/wallet/files.go

```diff
@@ -1,112 +1,80 @@
 package wallet
 
 import (
-	// "io/ioutil"
-	// "os"
+	"errors"
+	"io/ioutil"
+	"os"
 
 	"github.com/NebulousLabs/Sia/consensus"
 	"github.com/NebulousLabs/Sia/crypto"
-	// "github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/encoding"
 )
 
-// Key is how we serialize and store spendable addresses on disk.
-//
-// TODO: When recreating this file, perhaps there are more intelligent things
-// that can be done than just saving the keys. Otherwise at start the wallet
-// needs to do a full catchup on the state, which involves requesting the diffs
-// of every single block.
-type Key struct {
-	UnlockConditions consensus.UnlockConditions
+func fileExists(filename string) bool {
+	_, err := os.Stat(filename)
+	return !os.IsNotExist(err)
+}
+
+type savedKey struct {
 	SecretKey        crypto.SecretKey
+	UnlockConditions consensus.UnlockConditions
 }
 
+// save writes the contents of a wallet to a file.
 func (w *Wallet) save() (err error) {
-	/*
-		// Add every known spendable address + secret key.
-		var i int
-		keys := make([]Key, len(w.addresses))
-		for _, address := range w.addresses {
-			key := Key{
-				SpendConditions: address.spendConditions,
-				SecretKey:       address.secretKey,
-			}
-			keys[i] = key
-			i++
-		}
-
-		timelockedKeys := make([]AddressKey, len(w.timelockedAddresses))
-		for _, address := range w.timelockedAddresses {
-			key := Key{
-				SpendConditions: address.spendConditions,
-				SecretKey:       address.secretKey,
-			}
-			timelockedKeys[i] = key
-			i++
-		}
+	// convert key map to a slice
+	keySlice := make([]savedKey, len(w.keys))
+	for _, key := range w.keys {
+		keySlice = append(keySlice, savedKey{key.secretKey, key.unlockConditions})
```

Comment:
I think you meant `keySlice[i] = savedKey{...}`

Calling append will give you a doubly long slice where the first half is just blank.